numpy/_core/src/multiarray/datetime.c

```diff
@@ -2076,8 +2080,8 @@ NpyDatetime_ConvertPyDateTimeToDatetimeStruct(
     Py_DECREF(tmp);
 
     /* Get the month */
-    tmp = PyObject_GetAttrString(obj, "month");
-    if (tmp == NULL) {
+    res = PyObject_GetOptionalAttr(obj, npy_interned_str.month, &tmp);
```

Comment:
Instead of interning all these, can you use `PyObject_GetOptionalAttrString`?

Reply:
Hi @eric-wieser , Thank You for the feedback.

the interned version would help us get faster as we would be saving time, compare to the PyObject_GetOptionalAttrString which will be initialising the new string every time.